core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -655,6 +667,29 @@ public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings,
     }
 
     /**
+     * This method returns true if the current node is allowed to delete the given index's
+     * metadata folders only, but not the shard data. This is the case if the index is on a
+     * shared file system, and it is deleted without having been closed before, in which case
+     * the shared shard data should not be deleted because other nodes would have data structures
+     * depending on that shared data.
+     * @param index {@code Index} to check whether deletion is allowed
+     * @param indexSettings {@code IndexSettings} for the given index
+     * @param closed true if the index has been deleted closed already, but not yet deleted
+     * @return true if the index can be deleted on this node
+     */
+    public boolean canDeleteIndexMetadataOnly(Index index, IndexSettings indexSettings, boolean closed) {
```

Comment:
Then I think its not an issue, because we call `            MetaDataStateFormat.deleteMetaState(nodeEnv.indexPaths(index));` in a finally block anyway.  I'll revert out of these changes and just create the tests that manifest the problem for future work.